app/src/test/java/com/github/se/signify/model/common/user/FirestoreUserRepositoryTest.kt

```diff
@@ -982,4 +987,184 @@ class FirestoreUserRepositoryTest {
     // Verifies that the `update` operation was called with the expected data
     verify(mockCurrentUserDocRef).update(updatedDataCaptor.firstValue)
   }
+
+  @Suppress("UNCHECKED_CAST")
+  @Test
+  fun getFriendsList_shouldHandleUserDocNotExist() {
```

Comment:
We did not test if it handle when the user doc does not exist before, its a good thing to add those edge cases for every methods